API/CopilotService/__init__.py

```diff
@@ -77,31 +80,55 @@ async def _retrieve_context_from_learn_knowledge_index(question):
     system_msg = os.environ.get("OPENAI_SPLIT_TASK_MSG", default=DEFAULT_SPLIT_TASK_MSG)
     generate_results = gpt_generate(system_msg, question, history_msg=[])
     try:
-        task_list = _build_scenario_response(generate_results)
+        raw_task_list = _build_scenario_response(generate_results)
     except Exception as e:
         logger.error(f"Error while parsing the generate results: {generate_results}, {e}")
+        return None, None
 
-    splited_tasks = []
-    for task_info in task_list:
-        # use task command as task info when length of task info > 1, otherwise use task desc as task info
-        task_info = task_info.split("||")
-        task_info = task_info[1] if len(task_info) > 1 else task_info[0]
-        splited_tasks.append(asyncio.create_task(retrieve_chunks_for_atomic_task(task_info)))
+    context_tasks = [asyncio.create_task(_build_task_context(raw_task)) for raw_task in raw_task_list]
 
-    chunks_list = []
-    if len(splited_tasks) > 0:
-        chunks_list = await asyncio.gather(*splited_tasks)
+    context_infos = await asyncio.gather(*context_tasks)
+    task_list = [context_info[0] for context_info in context_infos]
+    chunk_list = [chunk for context_info in context_infos for chunk in context_info[1]]
 
+    chunk_list = merge_chunks_by_command(chunk_list)
     # TODO The logic of filtering, ranking, and aggregating chunks
 
-    return task_list, chunks_list
-
-
-def _add_context_to_queston(question, task_list, usage_context):
+    return task_list, chunk_list
+
+
+async def _build_task_context(raw_task):
+    desc = raw_task.split("||")[0]
+    if len(raw_task.split("||")) > 1:
+        cmd = raw_task.split("||")[1]
+        validate_result = validate_command_in_task(cmd)
+        if validate_result is None:
```

Comment:
Please determine whether this command is from CLI core or CLI extension

Reply:
I will add this logic in next Pull Request.